### ISSUE TITLE
Creation of dynamic property UcAddress::$nid is deprecated

### ISSUE THREAD

**Issue by herbdool** [backdrop-contrib/ubercart#517]
Error:

> Deprecated function: Creation of dynamic property UcAddress::$nid is deprecated in DatabaseStatementBase->fetchAllAssoc() (line 2314 of /app/httpdocs/core/includes/database/database.inc).

**Comment by argiepiano** [comment]
Hi Herb. Can you please provide more info? When are you seeing this?

**Comment by herbdool** [comment]
I'll have to think about that. I'm getting it on a site I've got in development that's running PHP 8.2.

**Comment by argiepiano** [comment]
Once you find out, let us know. I tried to reproduce locally with 8.2 without success

**Comment by herbdool** [comment]
@argiepiano I get it on `admin/store/products/view` after clearing cache. I do have uc_addresses enabled. I haven't compared to a site without that installed.

**Comment by argiepiano** [comment]
I'm still unable to reproduce, even with uc_addresses enabled. Any other details? What other contrib modules do you use?

**Comment by herbdool** [comment]
Looking into it more. I think it's got something to do with `uc_quote_node_load()` in shipping:

```
$addresses = db_query("SELECT nid, first_name, last_name, company, street1, street2, city, zone, postal_code, country, phone FROM {uc_quote_product_locations} WHERE nid IN (:nids)", array(':nids' => array_keys($nodes)), array('fetch' => 'UcAddress'))->fetchAllAssoc('nid');
```

I'm guessing `UcAddress` may have different ids depending on which table is implementing that. In `uc_quote_product_locations` it has `nid`, but in `uc_addresses` it has `aid` and `uid`. The easiest thing to do is just allow dynamic properties, but I'm not sure if you want to address this more fully.

**Comment by argiepiano** [comment]
Yes, I'm able to reproduce.  It happens when you create a product and enter a Product Pickup Address in the node creation. I think the quickest solution is to add `$nid` as a property of `UcAddress`. Would you be able to provide a PR?

**Comment by herbdool** [comment]
I can.

I had thought that `uc_addresses` was also using that class, but actually it appears not. It rolled its own approach to addresses. So adding `$nid` seems to be sufficient.

I didn't know that one could use `db_query` to fetch into a class. I don't see it used anywhere in core, except in `database_test.test`. Perhaps because most classes/entities can span multiple tables.

**Comment by herbdool** [comment]
@argiepiano I've got a PR. I took the approach of extending the `UcAddress` class and put the `nid` property there since it only relates to product shipping quotes and not addresses in general. I figured it would be clearer this way.

We'll see if tests pass.

**Comment by argiepiano** [comment]
> I didn't know that one could use db_query to fetch into a class. I don't see it used anywhere in core

Yes. It's used by [Entity](https://github.com/backdrop/backdrop/blob/f3a87ace9d9e54a2803a732a2d3ded3e662b2d49/core/modules/entity/entity.controller.inc#L345). This is how entity classes get created and populated when you load an entity.

**Comment by argiepiano** [comment]
The fix looks good to me. I added a tiny comment. No need to override `__construct()` since it's just calling the parent with the same parameter.

**Comment by argiepiano** [comment]
RTBC!

**Comment by herbdool** [comment]
@bugfolder @argiepiano this warning is bugging me. Can we get it merged?

**Comment by bugfolder** [comment]
Merged (along with a bunch of other RTBC PRs) and a new bugfix release created.

Sorry for the delay. My Ubercart cycles have been focused on https://github.com/backdrop-contrib/ubercart/issues/510.
